pilot/pkg/networking/util/util.go

```diff
@@ -407,6 +407,26 @@ func BuildConfigInfoMetadata(config config.Meta) *core.Metadata {
 	}
 }
 
+// AddConfigInfoMetadata adds name.namespace of the config, the type, etc
```

Comment:
why do we still have BuildConfigInfoMetadata?

Reply:
For simplicity I guess. Some object are built by calling BuildConfigInfoMetadata inline. which seems to be cleaner than calling an extra AddConfigInfoMetadata afterwards. I modified BuildConfigInfoMetadata a bit to make it call AddConfigInfoMetadata internally so that there is no code duplication.

Reply:
Let me know if you want me to remove Build and use Add. It does bring some benefit so that metadata won't be accidentally overwrite.